tests/unit/connector/test_confluent_kafka.py

```diff
@@ -702,3 +694,84 @@ def test_get_next_without_hmac(self):
         # output message is the same as the input message
         kafka_next_msg = kafka.get_next(1)
         assert kafka_next_msg == expected_event
+
+    def test_update_default_configuration_overwrites_default_options_in_nested_field(self):
+        default_config = {
+            "option": {"with": {"multiple": "layers", "foo": "bar"}},
+            "another": "option",
+        }
+        user_config = {"option": {"with": {"foo": "bi"}}}
+        expected_config = {
+            "option": {"with": {"multiple": "layers", "foo": "bi"}},
+            "another": "option",
+        }
+        config = deepcopy(TestConfluentKafkaFactory.valid_configuration)
+        kafka = ConfluentKafkaFactory.create_from_configuration(config)
+        new_config = kafka.update_default_configuration(default_config, user_config)
+        assert new_config == expected_config
+
+    def test_update_default_configuration_overwrites_default_options_in_first_level(self):
+        default_config = {
+            "option": {"with": {"multiple": "layers", "foo": "bar"}},
+            "another": "option",
+        }
+        user_config = {"another": "test option"}
+        expected_config = {
+            "option": {"with": {"multiple": "layers", "foo": "bar"}},
+            "another": "test option",
+        }
+        config = deepcopy(TestConfluentKafkaFactory.valid_configuration)
+        kafka = ConfluentKafkaFactory.create_from_configuration(config)
+        new_config = kafka.update_default_configuration(default_config, user_config)
+        assert new_config == expected_config
+
+    def test_update_default_configuration_raises_error_on_unknown_option_in_first_level(self):
+        default_config = {
+            "option": {"with": {"multiple": "layers", "foo": "bar"}},
+            "another": "option",
+        }
+        user_config = {"unknown": "option"}
+        config = deepcopy(TestConfluentKafkaFactory.valid_configuration)
+        kafka = ConfluentKafkaFactory.create_from_configuration(config)
+        with raises(UnknownOptionError, match="Unknown Option: unknown"):
+            _ = kafka.update_default_configuration(default_config, user_config)
+
+    def test_update_default_configuration_does_nothing_on_empty_user_configs(self):
+        default_config = {
+            "option": {"with": {"multiple": "layers", "foo": "bar"}},
+            "another": "option",
+        }
+        user_config = {}
+        config = deepcopy(TestConfluentKafkaFactory.valid_configuration)
+        kafka = ConfluentKafkaFactory.create_from_configuration(config)
+        new_config = kafka.update_default_configuration(default_config, user_config)
+        assert new_config == default_config
+
+    def test_update_default_configuration_raises_error_on_overwriting_all_subfields(self):
```

Comment:
This also seems to test for the type of field value to be replaced (as does the next test). Do we really need both?